asteroid/models/base_models.py

```diff
@@ -314,10 +315,7 @@ def forward(self, wav):
         decoded = self.postprocess_decoded(decoded)
 
         reconstructed = pad_x_to_y(decoded, wav)
-        if was_one_d:
-            return reconstructed.squeeze(0)
-        else:
-            return reconstructed
+        return reconstructed.squeeze(0) * was_one_d + reconstructed * (1 - was_one_d)
```

Comment:
Sorry for being late with the review. This is the only change that looks a bit weird to me. Was this necessary for Jit compat? If so, I think we should at least add a comment, or even better move this logic into a function with a descriptive name.

Reply:
I don't think this was necessary, @mhu-coder can you tell? 
Why does it look weird?

Reply:
It was necessary for tracing but as a traced model would have replaced `was_one_d` with a constant (with the value that was seen during tracing actually).
However, if we move this output shaping block in its own function, I am sure we could keep the old more readable logic and add the `torch.jit.scripting_if_tracing` decorator added to it. So the changes would be something like:

```python
def forward(self, wav):
   # ...
   return shape_output(reconstructed, wav.ndim)

@torch.jit.scripting_if_tracing
def shape_output(reconstructed, wav_dim):  # really not sure about the name of the function
    if wav_dim == 1:
        return reconstructed.squeeze(0)
    return reconstructed
```

Reply:
Sounds better. Mind submitting a PR?

Reply:
sure, on it

Reply:
Cool! It’s just a very uncommon way to write an if else clause :-) and also implicitly assumes 0/1 == True/False for Torch tensors